components/core/src/reducer/GroupByTimeOperator.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef REDUCER_GROUPBYTIMEOPERATOR_HPP
+#define REDUCER_GROUPBYTIMEOPERATOR_HPP
+
+#include <map>
+#include <string>
+
+#include "GroupTags.hpp"
+#include "Operator.hpp"
+#include "Record.hpp"
+
+namespace reducer {
+/**
+ * Operator that takes in records and forwards them down the pipeline with a group tag determined
+ * by their "@time" attribute.
+ *
+ * The group tag is created based on the configured "bucket size" in milliseconds which will align
+ * timestamps to n * "bucket size" milliseconds since epoch boundaries (taking the nearest boundary
+ * lower than the provided timestamp).
+ *
+ * TODO: the current implementation throws out all attributes of the record and its group tags which
+ * is fine for now because we only need it for local in memory group by -> count, but this operator
+ * should be generalized in the future.
+ */
+class GroupByTimeOperator : public Operator {
```

Comment:
Is this used anywhere? If not, can we get rid of it since it's confusing with the way we actually do group by time + count.

Reply:
I guess we can remove it